demo/list-box-presentation-demos.html

```diff
@@ -27,7 +27,7 @@ <h3>Separators</h3>
     <h3>Customizing listbox item label</h3>
     <vaadin-demo-snippet id="list-box-presentation-demos-custom-label">
       <template preserve-content>
-        <b>User</b>
+        <label>User</label>
```

Comment:
Label might not be semantically correct for this. Also see: https://github.com/vaadin/vaadin-notification-flow/pull/98

Reply:
@tomivirkki Looking at the HTML living standard https://html.spec.whatwg.org/multipage/forms.html#the-label-element  
or the relevant MDN page: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

It is pretty clear that a `<label>` "can be" associated with a "labelable element" (those which you listed, plus form-associated custom elements) but doesn't have to be.

> The `label` element represents a caption in a user interface. The caption can be associated with a specific form control, known as the `label` element's **labeled control**, either using the `for` attribute, or by putting the form control inside the `label` element itself.
>
> Except where otherwise specified by the following rules, a label element has no labeled control.

So I think it's perfectly fine in this kind of context. Another alternative would be one of `h1`-`h6` but in this case I think `<label>` is more semantically correct since it "represents a caption in a user interface" and not a heading for a section.

Reply:
https://github.com/vaadin/vaadin-notification-flow/pull/98 is a completely different use case since there it was used for "regular text" and not for a UI caption like here.

Reply:
See the discussion at https://stackoverflow.com/questions/12195052/is-it-allowed-to-use-label-tag-without-labeled-control

Reply:
I read that discussion (which is over 7 years old btw) but I still think it's perfectly valid to use label in this case. I find it weird that those answers in SO seem to think the answer is clearly a "No", but there seems to be no mention in the spec explicitly mentioning if it's ok or not recommended to do this.

At least from the latest version of the spec it clearly seems to be valid to have a label that is not associated with any element (see my quotes above) and the purpose is "represents a caption in a user interface".

Btw this usage is exactly the same as how all Vaadin components use label. E.g. in `vaadin-text-field` or `vaadin-custom-field` when you give it a label it's also just a standalone label that is not associated with any control. See https://github.com/vaadin/vaadin-text-field/blob/master/src/vaadin-text-field.html#L19 The only difference is that Vaadin components give the label an additional click handler which focuses the element we want. The label in those cases also has no labelable element as a child and there is no `for` attribute either.

If you inspect the `<label>` in e.g. text field or custom field demos `$0.control === null` but that would return the labelled control in a normal case when a label is associated with e.g. an input.

Reply:
Actually it might be nice to do this in the demo instead:
```html
<vaadin-custom-field label="User">
  <vaadin-list-box>
    ...
  </vaadin-list-box>
</vaadin-custom-field>
```

But as I see it that does not really change the label semantics compared to a simple `<label>` before the `vaadin-list-box` (since the label is internally used in a similar way). Also the demo is slightly easier to understand using only a native `label` as it is technically simpler without `vaadin-custom-field` as then you don't need to know about or understand what `vaadin-custom-field` does. Though `vaadin-custom-field` does also include Lumo styling for the label which is nice.

I'm fine with either using `vaadin-custom-field` for the label here or just a plain `<label>` as I initially suggested. Both have their pros.

Reply:
1. Currently `vaadin-custom-field`  does not support list-box as it does not have validation.

2. `<label>` would make sense if we had Form participation API support, which is not true.

IMO it's ok to use `<span>` as we are already doing elsewhere.

Reply:
@web-padawan Are you suggesting we should swap `<label>` for `<span>` in all components, or only for this demo? If not for the components, then why would it be different for this demo?

> IMO it's ok to use <span> as we are already doing elsewhere

Can you link to one usage of `<span>` in the kind of case you are referring to?

Reply:
> 1. Currently vaadin-custom-field does not support list-box as it does not have validation.

That's a good point, though not really relevant for this demo as value is not used, but I guess it would give a wrong impression one of the main usages of custom field is for using its value property. I was suggesting using it only to get the styled label here, but I guess that's not as good of an idea as I initially thought.

Reply:
> Are you suggesting we should swap <label> for <span> in all components, or only for this demo?

We can keep `<label>`, I just pointed out it does not make much sense to me.

Le's not spend too much time on this discussion though. I'm ok with any choice you make.

Reply:
> 2. `<label>` would make sense if we had Form participation support, which is not true.

If we supported that, then it would make sense to use `for` attribute or wrap the control with the label, but I was suggesting using it standalone just as a generic "caption in a user interface" without any explicit association with the list box (due to it not being a "field" and not supporting form participation). Though we could add some aria attribute to note that the list box is labelled by the label, but I think that's unnecessary for the demo and would just make it slightly more complex by adding more attributes.

Reply:
> Le's not spend too much time on this discussion though. I'm ok with any choice you make.

This is of course a small thing and not that important in scope of a single demo, but it would be nice to have some agreement/decision for this kind of case so we know the way we want to do it consistently in other demos when we encounter a similar situation (when there is no better way to show a label for a component e.g. doesn't internally implement a label feature).

Reply:
I don't want to insist forcing this any specific way if nobody else agrees with me. Just wanted to point out first the relevant parts of the spec and why I think it would be slightly better to use `<label>` here.

We can just keep it as `<b>User</b>` (and agree to use that consistently where needed) if I didn't convince either of you.

@tomivirkki What do you think (after above discussion)?

Reply:
The label is used internally in components like textfield without binding to a control, that's true. They are used as the target for `aria-labelledby`, but indeed it's not necessary for them to be `<label>` elements. Don't know the original reasoning for this either, but they are still component internal details and not visible for anyone unlike the demos. Can be changed imo also but don't know if it's worth it, maybe we could fix this for the Lit based components.

I don't have time for a thorough research but I've been under the impression that label is generally associated with controls, and if there's no control involved, I don't see a reason to change the existing code.

A bit of googling also reveals:
"HTML specifications do not make it mandatory to associate labels with controls, but Web Content Accessibility Guidelines (WCAG) 2.0 do." from https://stackoverflow.com/questions/33214093/is-it-good-practice-to-use-label-for-non-input-non-interactive-elements